app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt

```diff
@@ -38,7 +44,24 @@ sealed class Event {
     object OpenedPocketStory : Event()
     object DarkModeEnabled : Event()
 
-    val extras: Map<String, Any>?
+    // Interaction Events
+    data class SearchBarTapped(val source: Source) : Event() {
+        enum class Source { HOME, BROWSER }
+        override val extras: Map<String, String>?
```

Comment:
Just a heads up -- the `extras` map will become `Map<Enum, String>` in the next android-components release.  This lets us check that the keys are valid at compile time rather than run time.  Unfortunately, it's a breaking change to the API -- and you're our first external user (lucky you!)

https://github.com/mozilla-mobile/android-components/pull/2403

Reply:
I actually really like this change 👍